main/webapp/modules/core/scripts/views/data-table/cell-ui.js

```diff
@@ -96,17 +96,35 @@ DataTableCellUI.prototype._render = function() {
       nonstringSpan.className = 'data-table-value-nonstring';
       nonstringSpan.textContent = cell.v;
       divContent.appendChild(nonstringSpan);
-    } else if (URL.looksLikeUrl(cell.v)) {
-      var url = document.createElement('a');
-      url.textContent = cell.v;
-      url.setAttribute('href', cell.v);
-      url.setAttribute('target', '_blank');
-      divContent.appendChild(url);
     } else {
-      var span = document.createElement('span');
-      span.textContent = cell.v;
-      divContent.appendChild(span);
-    }
+      var arr = cell.v.split(" ");
+      var spanArr = [];
+      for (var i=0; i<arr.length; i++) {
+        if (URL.looksLikeUrl(arr[i])) {
+          if (spanArr.length != 0 ) {
+            var span = document.createElement('span');
+            span.textContent = spanArr.join(" ");
+            divContent.appendChild(span).appendChild(document.createTextNode('\u00A0'));
+            spanArr = [];
+          } var url = document.createElement('a');
+            url.textContent = arr[i];
+            url.setAttribute('href', arr[i]);
+            url.setAttribute('target', '_blank');
+            divContent.appendChild(url);
+            if (i == arr.length-1) {
+             divContent.appendChild(url)
+            } else {
+            divContent.appendChild(url).appendChild(document.createTextNode('\u00A0'));
+            }
+          } else {
+          spanArr.push(arr[i]);
+          } 
```

Comment:
Indentation needs cleanup - 2 space indents should be used for each level. Any good IDE should be able to do this for you automatically.

Reply:
yes working on it

Reply:
@tfmorris @kushthedude @singhakshita - I have updated https://github.com/OpenRefine/OpenRefine/issues/2519#issue-595223760 to give some Additional Context into the use case.  Please take a look.

Reply:
i have updated span test , typos fixed ,indentation fixed.  i guess we now can recognise all separate urls as mentioned by @thadguidry  , if any thing else is supposed to be done please let me know